k8s/cloud_deps/public/postgres/postgres_deployment.yaml

```diff
@@ -34,6 +39,46 @@ spec:
         - mountPath: /var/lib/postgresql/data
           subPath: data
           name: postgres-pv-claim
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+            - ALL
+          runAsNonRoot: true
+          runAsUser: 10100
+          seccompProfile:
+            type: RuntimeDefault
+      initContainers:
+      - name: init-chown-data
+        image: busybox
+        command: ["chown", "-R", "10100:10100", "/var/lib/postgresql/data"]
```

Comment:
At the start of the postgres container, a process called `initdb` get executed which will run `chown` on the `$PGDATA` which is by default set to `/var/lib/postgresql/data`. 

Postgres container by default is [built](https://github.com/docker-library/postgres/blob/master/14/alpine/Dockerfile#L159) to own the `/var/lib/postgresql/data` by postgres (uid - 70) user. 
<img width="533" alt="Screenshot 2023-02-08 at 14 59 07" src="https://user-images.githubusercontent.com/29277992/217498311-e4ea7fb9-725e-4f61-9af0-b67d763ce3f0.png">


But once a persistent volume get mounted to the path, the CSI driver[^1] set the uid of that mount point to root as shown below. Which in turns fails the `initdb`.
<img width="603" alt="Screenshot 2023-02-08 at 14 57 51" src="https://user-images.githubusercontent.com/29277992/217498388-4de1cf90-2c5c-4ac9-b7e8-a73839f43e4d.png">

The only way around this issue without giving the container the root access, is to run a temporary init container which will use root to change the ownership of the `data` directory within the persistent volume the UID we want. This is the [recommended workaround](https://learn.microsoft.com/en-us/troubleshoot/azure/azure-kubernetes/failure-setting-azure-disk-mount-options-uid-gid#workaround-2-use-an-init-container-and-the-change-owner-command-to-specify-the-user-and-group) by azure as well.

[^1]: We only tested this on AKS.